### PR TITLE
IGNITE-18533 Management API to kill client connection

### PR DESCRIPTION
**Pull request by nizhikov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Review by timoninmaxim — CHANGES_REQUESTED** [review]

File: modules/control-utility/src/test/java/org/apache/ignite/util/KillCommandsControlShTest.java [hunk 1]
```diff
@@ -153,6 +154,18 @@ public void testCancelContinuousQuery() throws Exception {
         });
     }
 
+    /** @throws Exception If failed. */
+    @Test
+    public void testCancelClientConnection() {
+        doTestCancelClientConnection(srvs, connId -> {
+            String nodeId = srvs.get(0).localNode().id().toString();
+
+            int res = execute("--kill", "client", "--node-id", nodeId, connId == null ? "ALL" : Long.toString(connId));
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/control-utility/src/test/java/org/apache/ignite/util/KillCommandsControlShTest.java:163]: > Should command failed if connection wasn't dropped

Yes. Fixed.

> Let's add test for specific node id also

Added.

File: docs/_docs/sql-reference/operational-commands.adoc [hunk 2]
```diff
@@ -397,3 +397,30 @@ control.bat --kill CONSISTENCY
...
+[source,bash]
+----
+control.bat --kill CLIENT connection_id [--node-id node_id]
+----
+
+--
+
+=== Parameters
+
+* `connection_id` - corresponds to the connection id of the client. Note, connection_id are local value.
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply docs/_docs/sql-reference/operational-commands.adoc:423]: Yes. Fixed.

File: modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java [hunk 3]
```diff
@@ -472,6 +479,40 @@ public static void doTestCancelContinuousQuery(IgniteEx cli, List<IgniteEx> srvs
...
+    public static void doTestCancelClientConnection(List<IgniteEx> srvs, Consumer<Long> cliCanceler) {
+        AtomicReference<IgniteClient> cli = new AtomicReference<>(
+            Ignition.startClient(new ClientConfiguration().setAddresses(Config.SERVER))
+        );
+
+        assertEquals(ClusterState.ACTIVE, cli.get().cluster().state());
+
+        List<List<?>> conns = execute(srvs.get(0), "SELECT CONNECTION_ID FROM SYS.CLIENT_CONNECTIONS");
+
+        assertEquals(1, conns.size());
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java]: Tests added.

File: docs/_docs/sql-reference/operational-commands.adoc [hunk 4]
```diff
@@ -397,3 +397,30 @@ control.bat --kill CONSISTENCY
...
+[source,bash]
+----
+control.bat --kill CLIENT connection_id [--node-id node_id]
+----
+
+--
+
+=== Parameters
+
+* `connection_id` - corresponds to the connection id of the client. Specify ALL to droll all connections. Note, connection_id are local value.
```
Comment (timoninmaxim) [inline docs/_docs/sql-reference/operational-commands.adoc:423]: s/droll/drop/

File: modules/core/src/main/java/org/apache/ignite/internal/sql/command/SqlKillClientCommand.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    /** Connections id. */
+    private Long connectionId;
+
+    /** {@inheritDoc} */
+    @Override public SqlCommand parse(SqlLexer lex) {
+        if (lex.shift()) {
+            if (lex.tokenType() == SqlLexerTokenType.DEFAULT) {
+                String connIdStr = lex.token();
+
+                if (!connIdStr.equals("ALL"))
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/sql/command/SqlKillClientCommand.java:41]: it's better to switch arguments - "ALL".equals(connIdStr).

**Review by timoninmaxim — APPROVED** [review]

**Pull request merged**